### PR TITLE
Ceph v14.2.8 is causing failure with OSDs on PVCs

### PR DESCRIPTION
**Pull request by leseb**
Ceph 14.2.8 just got released so let's use it.

Signed-off-by: Sébastien Han <seb@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

[test ceph]

### CODE REVIEW

**Comment by leseb** [comment]
arm64 packages are not there yet.

**Comment by leseb** [comment]
FAILURE: https://jenkins.rook.io/job/rook/job/rook/job/PR-4960/3/artifact/_output/tests/TestCephMultiClusterDeploySuite_aws_1.14.x_mrc-n1_rook-ceph-osd-0-7df4956d87-9glww_1583426694.log

**Comment by travisn** [comment]
@leseb Looks related to running on PVs, right?

**Comment by leseb** [comment]
> @leseb Looks related to running on PVs, right?

Yes! Only the PVC scenario.

**Comment by travisn** [comment]
I wonder if this only affects master since the `expand-bluefs` init container does not exist in the v1.2 releases.

**Comment by travisn** [comment]
There appear to be two issues:
1. The `expand-bluefs` init container is failing on v14.2.8. When this init container was removed as an experiment in #4970, the tests all succeeded except for the second bullet. Note that the init container would only affect master since it was not included in the v1.2 release.
1. The device is not being fully cleaned up between integration tests. The CephSmokeSuite fails in the osd prepare job with the following error:
```
2020-03-06 02:39:21.094438 I | cephosd: skipping device "xvdc": ["Has BlueStore device label"].
```

**Comment by travisn** [comment]
These issues only appear to affect the local PVs in the CI. These issues are not reproing with `cluster-on-pvc.yaml` in an AWS cluster.

File: pkg/operator/ceph/cluster/mon/spec.go [hunk 1]
```diff
@@ -183,9 +183,9 @@ func (c *Cluster) makeMonPod(monConfig *monConfig, canary bool, PVCName string)
 // Init and daemon containers require the same context, so we call it 'pod' context
 
 // PodSecurityContext detects if the pod needs privileges to run
-func PodSecurityContext() *v1.SecurityContext {
+func PodSecurityContext(force bool) *v1.SecurityContext {
```
Comment (travisn) [inline pkg/operator/ceph/cluster/mon/spec.go]: Only OSDs need to force to run privileged, right? Should we just have a helper method `privilegedContext()` that we only call in the OSD package? Then no need to modify all the other callers of this method.

Comment (travisn) [reply pkg/operator/ceph/cluster/mon/spec.go]: Or if you want to keep it this way, maybe rename the param to `forcePrivileged`.

Comment (leseb) [reply pkg/operator/ceph/cluster/mon/spec.go]: Yes, I feel that it might go broader than the OSD at some point, ok I'll rename to `forcePrivileged`.

Comment (travisn) [reply pkg/operator/ceph/cluster/mon/spec.go]: What about having a `PrivilegedContext()` method next to the `PodSecurityContext()`? So the OSD controller would call the new method and the the controllers would remain unchanged. Mmost callers use `!ForcePrivileged`, which is hard to understand IMO.

**Pull request merged**